src/views/Play/Play.tsx

```diff
@@ -729,8 +813,270 @@ export class Play extends React.Component<PlayProperties, any> {
+            <span className="head">{_("Organiser")}</span>
+            {/* <span className="head">{_("Rank")}</span> */}
+            <span className="head">{_("Size")}</span>
+            <span className="head time-control-header">{_("Time")}</span>
+            <span className="head">{_("")}</span>
+            <span className="head">{_("Handicap")}</span>
+            <span className="head" style={{textAlign: "left"}}>{_("Name")}</span>
+            <span className="head" style={{textAlign: "left"}}>{_("Rules")}</span>
+        </div>;
+    }
+
+    nominateForRengoChallenge = (C) => {
+        swal({
+            text: _("Joining..."),   // translator: the server is processing their request to join a rengo game
+            type: "info",
+            showCancelButton: false,
+            showConfirmButton: false,
+            allowEscapeKey: false,
+        }).catch(swal.noop);
+
+        put("challenges/%%/join", C.challenge_id, {})
+        .then(() => {
+            swal.close();
+        })
+        .catch((err) => {
+            swal.close();
+            errorAlerter(err);
+        });
+    };
+
+    unNominateForRengoChallenge = (C) => {
+        swal({
+            text: _("Withdrawing..."),   // translator: the server is processing their request to withdraw from a rengo challenge
+            type: "info",
+            showCancelButton: false,
+            showConfirmButton: false,
+            allowEscapeKey: false,
+        }).catch(swal.noop);
+
+        del("challenges/%%/join", C.challenge_id, {})
+        .then(() => {
+            swal.close();
+        })
+        .catch((err) => {
+            swal.close();
+            errorAlerter(err);
+        });
+    };
+
+    rengoList = () => {
+        if (!this.anyChallengesToShow(this.state.rengo_list)) {
+            return (
+                <div className="ineligible">
+                    {this.state.show_all_challenges ?
+                        _("(none)") /* translators: There are no challenges in the system, nothing to list here */ :
+                        _("(none available)") /* translators: There are no challenges that this person is eligible for */}
+                </div>
+            );
+        }
+
+        const user = data.get("user");
+
+        return this.state.rengo_list.map((C) => (
+            (((C.eligible || C.user_challenge || this.state.show_all_challenges)
+             && ((this.state.show_unranked_challenges && !C.ranked) || (this.state.show_ranked_challenges && C.ranked))
+             && ((this.state.show_19x19_challenges && C.width === 19 && C.height === 19) || (this.state.show_13x13_challenges && C.width === 13 && C.height === 13) || (this.state.show_9x9_challenges && C.width === 9 && C.height === 9) || (this.state.show_other_boardsize_challenges && (C.width !== C.height || (C.width !== 19 && C.width !== 13 && C.width !== 9)))) ) ?
+                    <div key={C.challenge_id} className={"challenge-row" + (C.rengo ? " rengo" : "")}>
```

Comment:
if `C` is a member of `rengo_list`, isn't it a given that it's a rengo challenge?

Reply:
Also, I looked in the stylesheets, I'm not sure this class name gets used

Reply:
I think that's a hangover from when I was toying with having Rengo games in amongst the other games